crates/sui-core/src/transaction_manager.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    sync::Arc,
+};
+
+use sui_storage::node_sync_store::NodeSyncStore;
+use sui_types::{
+    base_types::TransactionDigest, committee::EpochId, error::SuiResult,
+    messages::VerifiedCertificate,
+};
+use tokio::sync::mpsc::UnboundedSender;
+use tracing::error;
+
+use crate::authority::{authority_store::ObjectKey, AuthorityMetrics, AuthorityStore};
+
+/// TransactionManager is responsible for managing pending certificates and publishes a stream
+/// of certificates ready to be executed. It works together with AuthorityState for receiving
+/// pending certificates, and getting notified about committed objects. Executing driver
+/// subscribes to the stream of ready certificates published by the TransactionManager, and can
+/// execute them in parallel.
+/// TODO: use TransactionManager for fullnode.
+pub(crate) struct TransactionManager {
+    authority_store: Arc<AuthorityStore>,
+    node_sync_store: Arc<NodeSyncStore>,
+    missing_inputs: BTreeMap<ObjectKey, (EpochId, TransactionDigest)>,
+    pending_certificates: BTreeMap<(EpochId, TransactionDigest), BTreeSet<ObjectKey>>,
+    tx_ready_certificates: UnboundedSender<VerifiedCertificate>,
```

Comment:
I'm a bit worried about having a hidden, unbounded queue here, but I suppose its ok for now, especially since you are monitoring its size.

Eventually we need backpressure to allow us to reject user certs when we are overloaded, but that can be done later.